core/authentication_api.php

```diff
@@ -201,15 +201,18 @@ function auth_auto_create_user( $p_username, $p_password ) {
 
 	if( $t_login_method == BASIC_AUTH ) {
 		$t_auto_create = true;
+		$t_auto_create_password = md5( $p_password );
 	} else if( $t_login_method == LDAP && ldap_authenticate_by_username( $p_username, $p_password ) ) {
 		$t_auto_create = true;
+		$t_cache_password = config_get( 'ldap_cache_passwords' );
+		$t_auto_create_password = $t_cache_password ? md5( $p_password ) : 'external-password';
```

Comment:
Let's use auto-generated password (instead of 'external-password') so that the password can't be guessed if auth method is changed for example.  I would use a model like random cookie generation we already do.